test/conversionRate2.js

```diff
@@ -0,0 +1,1467 @@
+            assert.deepEqual(thisSell.valueOf(), baseSell[i].valueOf(), "wrong base sell rate.");
+        }
+    });
+
+    it("should set compact data all tokens then get and verify.", async function () {
+        //set compact data
+        compactBuyArr1 = [1, 2, 03, 04, 05, 06, 07, 08, 09, 10, 11, 12, 13, 14];
+        compactBuyArr2 = [15, 16, 17, 04, 05, 06, 07, 08, 09, 10, 11, 12, 13, 14];
+        let compactBuyHex = Helper.bytesToHex(compactBuyArr1);
+        buys.push(compactBuyHex);
+        compactBuyHex = Helper.bytesToHex(compactBuyArr2);
+        buys.push(compactBuyHex);
+
+        compactSellArr1 = [21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34];
+        compactSellArr2 = [35, 36, 37, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34];
+        let compactSellHex = Helper.bytesToHex(compactSellArr1);
+        sells.push(compactSellHex);
+        compactSellHex = Helper.bytesToHex(compactSellArr2);
+        sells.push(compactSellHex);
+
+        indices[0] = 0;
+        indices[1] = 1;
+
+        assert.deepEqual(indices.length, sells.length, "bad array size");
+        assert.deepEqual(indices.length, buys.length, "bad array size");
+
+        await convRatesInst.setCompactData(buys, sells, currentBlock, indices, {from: operator});
+        lastSetCompactBlock = currentBlock;
+
+        //get compact data for all tokens and verify as expected
+        for (i = 0; i < numTokens; ++i) {
+            let arrIndex = Math.floor (i / 14);
+            let fieldIndex = i % 14;
+            let compactResArr = await convRatesInst.getCompactData(tokens[i]);
+            let compactBuy;
+            let compactSell;
+
+            assert.equal(compactResArr[0].valueOf(), arrIndex.valueOf(), "wrong array " + i);
+            assert.equal(compactResArr[1].valueOf(), fieldIndex.valueOf(), "wrong field index " + i);
+            if (arrIndex == 0) {
+                compactBuy = compactBuyArr1;
+                compactSell = compactSellArr1;
+            }
+            else
+            {
+                compactBuy = compactBuyArr2;
+                compactSell = compactSellArr2;
+            }
+            assert.equal(compactResArr[2].valueOf(), compactBuy[fieldIndex].valueOf(), "wrong buy: " + i);
+            assert.equal(compactResArr[3].valueOf(), compactSell[fieldIndex].valueOf(), "wrong sell: " + i);
+        }
+
+        //get block number from compact data and verify
+        let blockNum = await convRatesInst.getRateUpdateBlock(tokens[3]);
+
+        assert.equal(blockNum.valueOf(), currentBlock.valueOf(), "bad block number returned");
+
+        blockNum = await convRatesInst.getRateUpdateBlock(tokens[11]);
+
+        assert.equal(blockNum.valueOf(), currentBlock.valueOf(), "bad block number returned");
+    });
+
+
+    it("should verify setCompactData reverted when block number out of range.", async function () {
+        let block = 0xffffffff1; //block number limited to size of int
```

Comment:
don't need this test here.
leave only tests relevant to code of this contract...